aws/data_source_aws_mks_cluster.go

```diff
@@ -0,0 +1,60 @@
+package aws
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsMksCluster() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsMksClusterRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"broker_node_group_info": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
I don't think this needs to be required if you are just passing the ARN to `readMskClusterState`